core/src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -1,10 +1,27 @@
 package pureconfig
 
+/**
+ * A mapping between case class fields and their respective keys in the config.
+ */
 trait ConfigFieldMapping extends (String => String) {
+
   def apply(fieldName: String): String
+
+  /**
+   * Returns a `ConfigFieldMapping` that uses this mapping with some exceptions.
+   *
+   * @param exceptions the exceptions for this mapping as pairs (field, configKey)
```

Comment:
@ruippeixotog It might be worth adding a comment explaining that any override will be applied globally throughout the configuration, e.g. if a `("foo", "bar")` exception is provided it will affect all fields named `foo` regardless of which object they're in and will affect all `bar` config keys regardless of the path leading up to the `.bar` part.